src/main.rs

```diff
@@ -353,7 +431,11 @@ async fn task(
 
                         for pixel in pixels.iter().copied() {
                             if x < display_width {
-                                display.set_pixel(x, y, pixel < 128);
+                                if pixel < 128 {
+                                    display.set_pixel(x as u16, y as u16, RgbColor::BLACK).unwrap(); 
+                                } else {
+                                    display.set_pixel(x as u16, y as u16, RgbColor::WHITE).unwrap(); 
+                                }
```

Comment:
On a color display we don't have to downgrade to 1-bit gray

```suggestion
                                display.set_pixel(x as u16, y as u16, Gray8::new(pixel).into()).unwrap();
```

(ofc we need some conditional compilation or `display`-module abstraction here, ssd1306 has a different API)